client_test.go

```diff
@@ -208,3 +210,20 @@ func TestUseFstatChecked(t *testing.T) {
 	testFstatOption(t, UseFstat(true), true)
 	testFstatOption(t, UseFstat(false), false)
 }
+
+type sink struct{}
+
+func (*sink) Close() error                { return nil }
+func (*sink) Write(p []byte) (int, error) { return len(p), nil }
+
+func TestBrokenServer(t *testing.T) {
+	// Simulate a server that sends a single packet with zero length,
+	// then closes the connection. This used to cause a panic in the client.
+
+	packet := []byte{0, 0, 0, 0}
+	w := &sink{}
+
+	r := bytes.NewReader(packet[:])
+	_, err := NewClientPipe(r, w)
+	assert.Equal(t, errShortPacket, err)
```

Comment:
I don’t think it’s necessary to check the type of the error, only that the operation did not succeed.

Reply:
See the discussion above about what error should be returned here.

Reply:
Thanks, i'll defer to the other maintainers, but I would prefer not to take another dependency for this check

Reply:
stretchr/testify is already a dependency, it just wasn't imported in this file. But I've removed it for now.

Reply:
I tend to aire on the side of not checking specific errors as well. It is too easy then for the error itself to become a part of your contract, and then it becomes hard to change without breaking tests.

Reply:
I've now changed the test to check for err == nil || err == io.EOF.